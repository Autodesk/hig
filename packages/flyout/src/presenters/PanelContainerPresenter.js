import React from "react";
import PropTypes from "prop-types";
import { css, cx } from "emotion";
import { ThemeContext } from "@hig/theme-context";

import stylesheet from "./stylesheet";
// import "./PanelContainerPresenter.scss";

export default function PanelContainerPresenter(props) {
  const { children, innerRef, maxHeight } = props;
  const maxHeightInPixels = maxHeight ? `${maxHeight}px` : undefined;

  return (
    <ThemeContext.Consumer>
      {({ resolvedRoles }) => {
        const styles = stylesheet({ transitionStatus: null, anchorPoint: null }, resolvedRoles);

        return (
          <div
            className={cx(css(styles.panelContainer), "hig__flyout-v1__panel-container")}
            ref={innerRef}
            style={{ maxHeight: maxHeightInPixels }}
          >
            <div className={cx(css(styles.panelContainerInner), "hig__flyout-v1__panel-container__inner")}>{children}</div>
          </div>
        );
      }}
    </ThemeContext.Consumer>
  );
}

PanelContainerPresenter.propTypes = {
  /** A required ref that's used to position the flyout */
  innerRef: PropTypes.func.isRequired,
  /** The panel content */
  children: PropTypes.node,
  /** Max height of the panel */
  maxHeight: PropTypes.number
};
