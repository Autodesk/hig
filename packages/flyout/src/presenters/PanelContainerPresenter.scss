.hig__flyout-v1__panel-container {
  @import "~@hig/styles/mixins/shadows.scss";
  @import "../tokens.scss";

  @include shadow("common");

  // border: $panel-border-width solid color(hig-cool-gray-20);
}
