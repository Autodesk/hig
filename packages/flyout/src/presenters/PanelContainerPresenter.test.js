import React from "react";
import renderer from "react-test-renderer";

import PanelContainerPresenter from "./PanelContainerPresenter";

describe("flyout/presenters/PanelContainerPresenter", () => {
  [
    {
      description: "renders without props",
      props: {}
    },
    {
      description: "renders with all props",
      props: {
        maxHeight: 150,
        innerRef: function innerRef() {},
        children: "Hello World"
      }
    }
  ].forEach(({ description, props: { children, ...otherProps } }) => {
    it(description, () => {
      const presenter = (
        <PanelContainerPresenter {...otherProps}>
          {children}
        </PanelContainerPresenter>
      );
      const tree = renderer.create(presenter).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
