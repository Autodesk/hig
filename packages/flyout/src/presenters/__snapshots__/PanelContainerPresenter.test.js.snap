// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flyout/presenters/PanelContainerPresenter renders with all props 1`] = `
.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  border: none;
  box-shadow: 0 0 16px rgba(0,0,0,0.2);
}

.emotion-0 {
  position: relative;
}

<div
  className="emotion-1"
  style={
    Object {
      "maxHeight": "150px",
    }
  }
>
  <div
    className="emotion-0"
  >
    Hello World
  </div>
</div>
`;

exports[`flyout/presenters/PanelContainerPresenter renders with only \`innerRef\` 1`] = `
.emotion-1 {
  background-color: #ffffff;
  border-radius: 4px;
  border: none;
  box-shadow: 0 0 16px rgba(0,0,0,0.2);
}

.emotion-0 {
  position: relative;
}

<div
  className="emotion-1"
  style={
    Object {
      "maxHeight": undefined,
    }
  }
>
  <div
    className="emotion-0"
  />
</div>
`;
